### PR TITLE
[WIP] CNF-6019 Improve TALM conditions, reasons, and messages

### PR DESCRIPTION
**Pull request by fontivan**
This is still WIP but most of the basic states should be working. README/docs still need to be updated.
Significant retesting required (mostly creating this PR so I can get a CI run)

### CODE REVIEW

File: controllers/backup.go [hunk 1]
```diff
@@ -232,34 +233,67 @@ func (r *ClusterGroupUpgradeReconciler) backupActive(ctx context.Context, cluste
...
+				metav1.ConditionTrue,
+				"Backup failed for all clusters",
+			)
+		} else {
+			utils.SetStatusCondition(
+				&clusterGroupUpgrade.Status.Conditions,
+				utils.ConditionTypes.BackupSuceeded,
+				utils.ConditionReasons.PartiallyDone,
+				metav1.ConditionTrue,
+				fmt.Sprintf("Backup failed for clusters: [%s]", strings.Join(failedBackupClusters, ",")),
```
Comment (sabbir-47) [inline controllers/backup.go:286]: For me, the message for which cluster failed is not necessary, suppose, if we launch the CU having 100 clusters, and in the worst case, all the cluster failed, we are not going to overload the message variable with tons of strings.

**Comment by fontivan** [comment]
Superceded by https://github.com/openshift-kni/cluster-group-upgrades-operator/pull/311

**Pull request closed without merging**